gog/graph.go

```diff
@@ -274,5 +272,8 @@ func (g *Grapher) makeDefInfo(obj types.Object) (*DefKey, *defInfo) {
 		path = append([]string{filepath.Base(p.Filename)}, path...)
 	}
 
-	return &DefKey{obj.Pkg().Path(), path}, &defInfo{pkgscope: g.pkgscope[obj], exported: g.exported[obj]}
+	return &DefKey{obj.Pkg().Path(), path}, &defInfo{
+		pkgscope: g.pkgscope[obj],
+		exported: obj.Exported() && (obj.Parent() == nil || obj.Parent() == obj.Pkg().Scope()),
```

Comment:
I'm trusting that the test coverage will catch if this check is exhaustive enough. But from reading it, it sounds like it. Maybe a tiny helper function with a descriptive name?